src/GUIObject.cpp

```diff
@@ -228,8 +218,15 @@ void GUIObject::setSelectedControl(int index) {
 			if (dynamic_cast<GUIButton*>(obj) || dynamic_cast<GUICheckBox*>(obj)) {
 				//It's a button.
 				obj->state = (i == index) ? 1 : 0;
-			} else if (dynamic_cast<GUITextBox*>(obj) || dynamic_cast<GUISpinBox*>(obj)) {
+			} else if (dynamic_cast<GUITextBox*>(obj)) {
 				//It's a text box.
+				if(i == index) {
+					obj->state = 2;
+				} else {
+					dynamic_cast<GUITextBox*>(obj)->blur();
+				}
+			} else if (dynamic_cast<GUISpinBox*>(obj)) {

```

Comment:
I forgot that the GUISpinBox is already a subclass of GUITextBox now, so maybe you can remove some of the GUISpinBox type check in this file?